### PR TITLE
Trit 98

### PR DESCRIPTION
**Pull request by apodi**
TRIT-98 Full Time Overseas Work in Tax Year SCREEN

### CODE REVIEW

File: conf/app.routes [hunk 1]
```diff
@@ -39,3 +39,9 @@ POST       /days-spent-in-UK                            controllers.DaysSpentInU
 GET        /change-days-spent-in-UK                     controllers.DaysSpentInUKController.onPageLoad(mode: Mode = CheckMode)
 
 POST       /change-days-spent-in-UK                     controllers.DaysSpentInUKController.onSubmit(mode: Mode = CheckMode)
+
+
+GET        /full-time-overseas-work                        controllers.FullTimeOverseasWorkController.onPageLoad(mode: Mode = NormalMode)
```
Comment (abhijeetardale) [inline conf/app.routes]: please check the alignments

File: test/views/FullTimeOverseasWorkSpec.scala [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+class FullTimeOverseasWorkSpec extends SpecBase with MockitoSugar with NunjucksSupport with
+  JsonMatchers with ViewBehaviours with QuestionViewBehaviours{
+
+  lazy val fullTimeOverseasWorkRoute: String = routes.FullTimeOverseasWorkController.onPageLoad(NormalMode).url
+  implicit val request = FakeRequest(GET, fullTimeOverseasWorkRoute).withCSRFToken
+  val formProvider = new FullTimeOverseasWorkFormProvider()
+  val form = formProvider()
+  val html: Html = renderer.render("fullTimeOverseasWork.njk").futureValue
+
+  behave like normalPageWithDyanamicTitle(html, "fullTimeOverseasWork","hint")
```
Comment (abhijeetardale) [inline test/views/FullTimeOverseasWorkSpec.scala]: view tests for errors are missing

File: app/controllers/FullTimeOverseasWorkController.scala [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    val controllerComponents: MessagesControllerComponents,
+    renderer: Renderer
+)(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport with NunjucksSupport {
+
+  private val form = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData ).async {
+    implicit request =>
+
+      val preparedForm = request.userAnswers.get(FullTimeOverseasWorkPage) match {
```
Comment (abhijeetardale) [inline app/controllers/FullTimeOverseasWorkController.scala]: Is this page dependents upon any other page? if yes check is missing as data check

File: test/views/FullTimeOverseasWorkSpec.scala (hunk 2 above)
Comment (alsahid1993) [reply test/views/FullTimeOverseasWorkSpec.scala]: Testing for errors have been added

File: app/controllers/FullTimeOverseasWorkController.scala (hunk 3 above)
Comment (alsahid1993) [reply app/controllers/FullTimeOverseasWorkController.scala]: This class been updated. 
Currently waiting for review.

File: conf/app.routes (hunk 1 above)
Comment (alsahid1993) [reply conf/app.routes]: This class been updated. 
Currently waiting for review.

**Pull request merged**